File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-0.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        dp = {}
        dp[0] = 0
        dp[1] = 0
        for i in range(2, len(cost)+1):
            s1 = i - 1
            s2 = i - 2
            dp[i] = min(dp[s1] + cost[s1], dp[s2] + cost[s2])
        return dp[len(cost)]

File: Data Structures & Algorithms/jump-game-ii/submission-0.go

func jump(nums []int) int {
    dp := make(map[int]int);
    for i := 0 ; i < len(nums); i++ {
        dp[i] = math.MaxInt;
    }
    dp[0] = 0;
    for i, value := range nums {
        for s := 1; s<= value && i+s <= len(nums) - 1; s++ {
            j := i + s;
            dp[j] = min(dp[j], dp[i]+1);
        }
    }
    return dp[len(nums)-1];
}
